File: Problem 3.py
from __future__ import print_function, division
import math
import sys

# Borrowed from scipy/lib/six.py (sort of)
if sys.version_info.major == 2:
    import __builtin__
    builtins = __builtin__
    xrange = __builtin__.xrange
else:
    import builtins
    xrange = builtins.range


def isprime(num):
    if num == 2:
        return True
    bound = int(math.ceil(math.sqrt(num)))
    for cfactor in xrange(2, bound):
        if num % cfactor == 0:
            return False
    return True

if __name__ == '__main__':
    N = 600851475143
    factors = [
            [f, N // f] 
            for f in xrange(3,int(math.ceil(math.sqrt(N))))
            if (N % f) == 0
            ]
    factors = sum(factors, [])
    factors = [f for f in factors if isprime(f)]
    print("The answer to Euler Probelm 3 is", max(factors))

File: Problem 3.c
#include <stdio.h>
#include <math.h>


#ifndef N
// N=600851475143
#endif

int isprime(long n){
	if (n & 1){
		return 1;
	} 
	for (long i = 3; i < sqrt(n); ++i)
	{
		if (n % i == 0){
			return 0;
		} 
	}
	return 1;
}


int main(int argc, char const *argv[])
{
	long max_factor = 0;
	long N=600851475143;

	for (long i = 3; i < sqrt(N); ++i)
	{
		if (N % i == 0){
			if (isprime(i)){
				max_factor = i;
			} else if (isprime(N/i)){
				max_factor = N/i;
			}
		}
	}
	printf("The answer to Euler Problem 3 is %lu", max_factor);
}